### PR TITLE
EVG-6387 Add JSON output to list-patches as a flag

### PR DESCRIPTION
**Pull request by chasinglogic**
Sample output:

```
@raza ➜  evergreen git:(master) ./clients/linux_amd64/evergreen list-patches --json -n 1
[
	{
		"patch_id": "5d1676971e2d1735ef152247",
		"description": "fix-lint: SERVER-41168 Add docstrings to lldb debug extensions",
		"project_id": "mongodb-mongo-master",
		"branch": "mongodb-mongo-master",
		"git_hash": "7f2630cf028f38d45dd5d9e3415d44f31b3cc030",
		"patch_number": 851,
		"author": "mathew.robinson",
		"version": "5d1676971e2d1735ef152247",
		"status": "succeeded",
		"create_time": "2019-06-28T20:20:39.577Z",
		"start_time": "2019-06-28T20:20:39.577Z",
		"finish_time": "2019-06-28T20:20:39.577Z",
		"builds": [
			"enterprise-rhel-62-64-bit"
		],
		"tasks": [
			"lint"
		],
		"variants_tasks": [
			{
				"name": "enterprise-rhel-62-64-bit",
				"tasks": [
					"lint"
				]
			}
		],
		"activated": true,
		"alias": "",
		"github_patch_data": {
			"pr_number": 0,
			"base_owner": "",
			"base_repo": "",
			"head_owner": "",
			"head_repo": "",
			"head_hash": "",
			"author": ""
		}
	}
]
@raza ➜  evergreen git:(master) 
```

### CODE REVIEW

File: operations/patch_list.go [hunk 1]
```diff
@@ -64,13 +72,32 @@ func PatchList() cli.Command {
 				}
 			}
 
+			if outputJSON {
+				display := make([]restModel.APIPatch, len(patches)-1)
+
+				for _, p := range patches {
+					api := restModel.APIPatch{}
+					api.BuildFromService(p)
```
Comment (john-m-liu) [inline operations/patch_list.go]: This error needs to be handled (you can probably just return a wrapped copy of it)

Comment (chasinglogic) [reply operations/patch_list.go]: My bad, didn't realize that even returned an error, I've made this change.

Thanks!

File: operations/patch_list.go [hunk 2]
```diff
@@ -64,13 +72,35 @@ func PatchList() cli.Command {
 				}
 			}
 
+			if outputJSON {
+				display := make([]restModel.APIPatch, len(patches)-1)
```
Comment (tychoish) [inline operations/patch_list.go]: I think you want to set the size of the slice to 0, and the capacity (third argument to make) to len(patches). 

Though, frankly, I think for cases like this, I think making a slice without a size, and just using append is fine (and make in cases like this feels like a premature optimization.) 

Out of curiosity, though I don't get why you'd want len-1?

Comment (chasinglogic) [reply operations/patch_list.go]: Because I was misusing make and it gave me a bad result.

Comment (chasinglogic) [reply operations/patch_list.go]: I have made this change.

**Pull request merged**